docs/design/rubygems.md

```diff
@@ -0,0 +1,522 @@
+repository that has references to any platform besides `RUBY`.
+
+To achieve this, we can implement a check in Cachi2 that will cause the request to fail if any other platform is
+specified in the `Gemfile.lock`. This way, we can avoid downloading any pre-compiled content.
+
+We should probably implement support for pre-compiled binaries as a follow-up feature, similarly to what was done with
+[pip wheels](https://github.com/containerbuildsystem/cachi2/blob/main/docs/pip.md#distribution-formats).
+
+#### Checksum verification
+Since checksums in the `Gemlock.file` is still a feature in development (see [checksums](#dependency-checksums)), we
+can postpone implementing support for it until the feature is delivered.
+
+We need to decide if we will report all dependencies as having missing checksums in the SBOM, or not.
+
+#### Alternate registries
+Users can specify different registries by leveraging the `source` keyword on the `Gemfile`. This setting can be global
+or target to a specific group of Gems. See the [official docs](https://bundler.io/guides/gemfile.html#gemfiles) for
+more info.
+
+We can likely just support the use case of the global `source` being set to `'https://rubygems.org'` (the default)
+value. Cachi2 will need to check for ny changes on the global `source` or additional `source` blocks added to the
+`Gemfile`, and fail the request in case they're found.
+
+Support for alternate registries should be added as a follow-up story.
+
+#### Dev dependencies
+Bundler does not have the specific concept of development dependencies, instead relying on
+[groups](#dependency-groups-or-how-bundler-deals-with-dev-dependencies) to implement it in a more loose way.
+
+We could still consider that Gems under a group called `:development` are dev dependencies, but for this purpose,
+parsing the `Gemfile.lock` will not provide enough data. It seems that all Gems are placed in the `Gemfile.lock`
+without any information on groups.
+
+If this feature is really needed, we'd also need to parse the `Gemfile`. It can be done as a follow-up if the need
+arises.
+
+### Providing the content for the hermetic build
+
+#### Setting the Bundler configuration
+
+The order of precedence for Bundler configuration options is as follows:
+
+1. Local config (`<project_root>/.bundle/config or $BUNDLE_APP_CONFIG/config`)
+2. Environment variables (ENV)
+3. Global config (`~/.bundle/config`)
+4. Bundler default config
+
+Since the local configuration takes higher precedence than the environment variables (except `BUNDLE_APP_CONFIG`), we
+need to set the Bundler configuration options to make the build work.
+
+This can be achieved by using Cachi2's `inject-files` command to add the necessary keys to the local configuration,
+while respecting the other values that might exist there. In case the local config file does not exist, it needs to be
+created by this command.
```

Comment:
Could we `inject-files` the config into `${output_dir}/deps/rubygems/config` and set `BUNDLE_APP_CONFIG=${output_dir}/deps/rubygems` to avoid writing directly into the repo?

Reply:
(actually, maybe not - to support two or more bundler projects in one repo, I think we'd have to overwrite the individual files)